napari/utils/translations.py

```diff
@@ -0,0 +1,334 @@
+"""
+Localization utilities to find available language packs and packages with
+localization data.
+"""
+
+import gettext
+import importlib
+import os
+from pathlib import Path
+
+import babel
+
+# Entry points
+JUPYTERLAB_LANGUAGEPACK_ENTRY = "napari.languagepack"
+
+# Constants
+DEFAULT_LOCALE = "en"
+LOCALE_DIR = "locale"
+
+
+def _get_display_name(
+    locale: str, display_locale: str = DEFAULT_LOCALE
+) -> str:
+    """
+    Return the language name to use with a `display_locale` for a given language locale.
+    Parameters
+    ----------
+    locale: str
+        The language name to use.
+    display_locale: str, optional
+        The language to display the `locale`.
+    Returns
+    -------
+    str
+        Localized `locale` and capitalized language name using `display_locale` as language.
+    """
+    locale = locale if _is_valid_locale(locale) else DEFAULT_LOCALE
+    display_locale = (
+        display_locale if _is_valid_locale(display_locale) else DEFAULT_LOCALE
+    )
+    loc = babel.Locale.parse(locale)
+    return loc.get_display_name(display_locale).capitalize()
+
+
+def _is_valid_locale(locale: str) -> bool:
+    """
+    Check if a `locale` value is valid.
+
+    Parameters
+    ----------
+    locale: str
+        Language locale code.
+
+    Notes
+    -----
+    A valid locale is in the form language (See ISO-639 standard) and an
+    optional territory (See ISO-3166 standard).
+
+    Examples of valid locales:
+    - English: DEFAULT_LOCALE
+    - Australian English: "en_AU"
+    - Portuguese: "pt"
+    - Brazilian Portuguese: "pt_BR"
+
+    Examples of invalid locales:
+    - Australian Spanish: "es_AU"
+    - Brazilian German: "de_BR"
+    """
+    valid = False
+    try:
+        babel.Locale.parse(locale)
+        valid = True
+    except babel.core.UnknownLocaleError:
+        pass
+    except ValueError:
+        pass
+
+    return valid
+
+
+# --- Translators
+# ----------------------------------------------------------------------------
+class TranslationBundle:
+    """
+    Translation bundle providing gettext translation functionality.
+    """
+
+    def __init__(self, domain: str, locale: str):
+        self._domain = domain
+        self._locale = locale
+
+        self._update_locale(locale)
+
+    def _update_locale(self, locale: str):
+        """
+        Update the locale environment variables.
+
+        Parameters
+        ----------
+        locale: str
+            The language name to use.
+        """
+        self._locale = locale
+        localedir = None
+        if locale.split("_")[0] != DEFAULT_LOCALE:
+            language_pack_module = f"napari_language_pack_{locale}"
+            try:
+                mod = importlib.import_module(language_pack_module)
+                # FIXME: Check this is right
+                localedir = Path(mod.__file__).parent / LOCALE_DIR
```

Comment:
There is plan to napari team grab all possible combinations?

Reply:
Hi @Czaki I do not follow what you mean :)

Reply:
I think about hostile takeover name on pypi, but backage on pypi could have a different name than installed package.

Reply:
True, will move to use entry points, that way is safer :) 

Good catch!

Reply:
Fixed